src/modules/launcher/PowerLauncher/MainWindow.xaml.cs

```diff
@@ -291,12 +331,49 @@ private void AutoSuggestBox_QuerySubmitted(AutoSuggestBox sender, AutoSuggestBox
             }
         }
 
-        private void QueryTextBox_TextChanged(Windows.UI.Xaml.Controls.AutoSuggestBox sender, Windows.UI.Xaml.Controls.AutoSuggestBoxTextChangedEventArgs args)
+        private const int millisecondsToWait = 200;
+        private static DateTime s_lastTimeOfTyping;
+
+        private void QueryTextBox_TextChanged(object sender, Windows.UI.Xaml.Controls.TextChangedEventArgs e)
         {
-            if (args.Reason == AutoSuggestionBoxTextChangeReason.UserInput)
+            var latestTimeOfTyping = DateTime.Now;
+            var text = ((Windows.UI.Xaml.Controls.TextBox)sender).Text;
+            Task.Run(() => DelayedCheck(latestTimeOfTyping, text));
+            s_lastTimeOfTyping = latestTimeOfTyping;
+        }
+
+        private async Task DelayedCheck(DateTime latestTimeOfTyping, string text)
+        {
+            await Task.Delay(millisecondsToWait);
+            if (latestTimeOfTyping.Equals(s_lastTimeOfTyping))
             {
-                _viewModel.QueryText = sender.Text;
+                await System.Windows.Application.Current.Dispatcher.BeginInvoke(new Action(() =>
+                 {
+                     _viewModel.QueryText = text;
+                 }));               
             }
         }
+
+        private void WindowsXamlHost_PreviewMouseDown(object sender, MouseButtonEventArgs e)
+        {
+            //    if (sender != null && e.OriginalSource != null)
+            //    {
+            //        //var r = (ResultListBox)sender;
+            //        //var d = (DependencyObject)e.OriginalSource;
+            //        //var item = ItemsControl.ContainerFromElement(r, d) as ListBoxItem;
+            //        //var result = (ResultViewModel)item?.DataContext;
+            //        //if (result != null)
+            //        //{
+            //        //    if (e.ChangedButton == MouseButton.Left)
+            //        //    {
+            //        //        _viewModel.OpenResultCommand.Execute(null);
+            //        //    }
+            //        //    else if (e.ChangedButton == MouseButton.Right)
+            //        //    {
+            //        //        _viewModel.LoadContextMenuCommand.Execute(null);
+            //        //    }

```

Comment:
We can probably delete this, unless you think we'd be adding it back in.

Reply:
I don't think it will be required later since we have already handled this functionality. Thanks for pointing out.